content/en/blog/2024/new-otel-features-envoy-istio/index.md

```diff
@@ -0,0 +1,243 @@
+resource detector to be easily added with Envoy's built-in extensions feature.
+
+### Custom samplers
+
+Another exciting feature added to Envoy is the possibility of implementing and
+configuring custom samplers. Envoy follows the
+[OTel Sampler interface](/docs/specs/otel/trace/sdk/#sampler), which makes it
+easy for anyone to contribute their own samplers.
+
+Envoy ships with the
+[Always On Sampler](https://www.envoyproxy.io/docs/envoy/v1.29.4/api-v3/extensions/tracers/opentelemetry/samplers/v3/always_on_sampler.proto)
+which simply forwards all spans. This base implementation can be used as a
+reference implementation for smarter samplers.
+
+## Demo
+
+It's time to see the new features in action! For this, we will use the
+[Istio Bookinfo application](https://istio.io/latest/docs/examples/bookinfo/).
+We will deploy it in Kubernetes and use Istio as our service mesh. Traces will
+be exported to [Jaeger](https://www.jaegertracing.io/) using HTTP.
+
+### Install Jaeger
+
+First, start by installing the Jaeger operator:
+
+```shell
+kubectl create namespace observability
+kubectl create -f https://github.com/jaegertracing/jaeger-operator/releases/download/v1.57.0/jaeger-operator.yaml -n observability
+```
+
+Then deploy Jaeger `all-in-one`:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: jaegertracing.io/v1
+kind: Jaeger
+metadata:
+  name: simplest
+EOF
+```
+
+### Install and configure Istio
+
+Now, let's install Istio using
+[`istioctl`](https://istio.io/latest/docs/setup/install/istioctl/)
+
+```shell
+cat <<EOF | istioctl install -y -f -
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    enableTracing: true
+    extensionProviders:
+    - name: otel-tracing
+      opentelemetry:
+        port: 4318
+        service: simplest-collector.default.svc.cluster.local
+        http:
+          path: "/v1/traces"
+          timeout: 5s
+        resource_detectors:
+          environment: {}
+EOF
+```
+
+This installs Istio and configures the OpenTelemetry tracing provider. We are
+using the `http` exporter, and sending to the OTLP/HTTP endpoint in the Jaeger
+collector. We are also enabling the environment resource detector in
+`resource_detectors`.
+
+Next, we need to enable the tracer using Istio's
+[Telemetry API](https://istio.io/latest/docs/tasks/observability/telemetry/):
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: telemetry.istio.io/v1alpha1
+kind: Telemetry
+metadata:
+  name: otel-demo
+spec:
+  tracing:
+  - providers:
+    - name: otel-tracing
+    randomSamplingPercentage: 100
+EOF
+```
+
+And finally, we configure the `OTEL_RESOURCE_ATTRIBUTES` environment variable
+for the Envoy proxies:
+
+```shell
+cat <<EOF | k apply -f -
+apiVersion: networking.istio.io/v1beta1
+kind: ProxyConfig
+metadata:
+  name: my-proxyconfig
+  namespace: istio-system
+spec:
+  concurrency: 0
+  environmentVariables:
+    OTEL_RESOURCE_ATTRIBUTES: "host-name=abc-123"
```

Comment:
```suggestion
    OTEL_RESOURCE_ATTRIBUTES: "host-name=abc-123"
```

`host-name` is not covered by the semantic conventions, it should be `host.name`, I guess?

Reply:
Good catch, I wanted to add an arbitrary attribute but somehow picked host :D. Yes, `host.name` is the semconv correct key. Fixed!